### PR TITLE
Adds list and search methods for issues

### PR DESCRIPTION
**Pull request by fedefernandez**
This PR adds the list and search methods for issues

Please @juanpedromoreno could you review? I'm especially interested in your opinion with the definition of [`SearchParam`](https://github.com/47deg/github4s/compare/search-and-list-issues?expand=1#diff-a8db947de03bf1b9d71de7305abeb5bb)

Thanks!

### CODE REVIEW

File: github4s/js/src/test/scala/github4s/utils/integration/GHIssuesSpec.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    })
+  }
+
+  "Issues >> Search" should "return at least one issue for a valid query" in {
+    val response = Github(accessToken).issues
+      .searchIssues(validSearchQuery, validSearchParams)
+      .execFuture(headerUserAgent)
+
+    testFutureIsRight[SearchIssuesResult](response, { r =>
+      r.result.total_count > 1 shouldBe true
```
Comment (juanpedromoreno) [inline github4s/js/src/test/scala/github4s/utils/integration/GHIssuesSpec.scala]: This assert is saying that at least should exist two issues, which is incongruent with the test banner `return at least one issue for a valid query`

File: github4s/jvm/src/test/scala/github4s/integration/GHIssuesSpec.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  "Issues >> Search" should "return at least one issue for a valid query" in {
+    val response = Github(accessToken).issues
+      .searchIssues(validSearchQuery, validSearchParams)
+      .exec[Id, HttpResponse[String]](headerUserAgent)
+
+    response should be('right)
+    response.toOption map { r ⇒
+      r.result.total_count > 1 shouldBe true
```
Comment (juanpedromoreno) [inline github4s/jvm/src/test/scala/github4s/integration/GHIssuesSpec.scala]: Same here.

File: github4s/shared/src/main/scala/github4s/api/Issues.scala [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    * Note: In the past, pull requests and issues were more closely aligned than they are now.
+    * As far as the API is concerned, every pull request is an issue, but not every issue is a
+    * pull request.
+    *
+    * This endpoint may also return pull requests in the response. If an issue is a pull request,
+    * the object will include a `pull_request` key.
+    *
+    * @param accessToken to identify the authenticated user
+    * @param headers optional user headers to include in the request
+    * @param searchParams list of search params
```
Comment (juanpedromoreno) [inline github4s/shared/src/main/scala/github4s/api/Issues.scala]: Let's include what `query` parameter does too, to complete the scaladoc ;)

**Comment by fedefernandez** [comment]
Ready for a second review!

Thanks @juanpedromoreno

**Comment by juanpedromoreno** [comment]
LGTM! Thanks @fedefernandez !

**Pull request merged**